HARK/ConsumptionSaving/ConsIndShockModel_AgentDicts.py

```diff
@@ -0,0 +1,443 @@
+# -*- coding: utf-8 -*-
+from copy import copy, deepcopy
+from HARK.datasets.life_tables.us_ssa.SSATools import parse_ssa_life_table
+from HARK.datasets.SCF.WealthIncomeDist.SCFDistTools \
+    import income_wealth_dists_from_scf
+from HARK.Calibration.Income.IncomeTools \
+    import (parse_income_spec, parse_time_params, Cagetti_income)
+
+"""
+Define parameters for various Consumer AgentTypes
+"""
+
+init_perfect_foresight_plus = {}
+
+# The info below is optional at present but may become mandatory as the toolkit evolves
+# 'Primitives' define the 'true' model that we think of ourselves as trying to solve
+# (the limit as approximation error reaches zero)
+
+init_perfect_foresight_plus.update(
+    {'prmtv_par': ['CRRA', 'Rfree', 'DiscFac', 'LivPrb', 'PermGroFac', 'BoroCnstArt', 'PermGroFacAgg', 'T_cycle', 'cycles']})
+# Approximation parameters define the precision of the approximation
+# Limiting values for approximation parameters: values such that, as all such parameters approach their limits,
+# the approximation gets arbitrarily close to the 'true' model
+init_perfect_foresight_plus.update(  # In principle, kinks exist all the way to infinity
+    {'aprox_par': {'MaxKinks': '500'}})
+init_perfect_foresight_plus.update(  # In principle, kinks exist all the way to infinity
+    {'aprox_lim': {'MaxKinks': float('inf')}})
+
+# The simulation stge of the problem requires additional parameterization
+init_perfect_foresight_plus.update(  # The 'primitives' for the simulation
+    {'prmtv_sim': ['aNrmInitMean', 'aNrmInitStd', 'pLvlInitMean', 'pLvlInitStd']})
+init_perfect_foresight_plus.update({  # Approximation parameters for monte carlo sims
+    'sim_mcrlo': ['AgentCount', 'T_age']
+})
+init_perfect_foresight_plus.update({  # Limiting values that define 'true' simulation
+    'sim_mcrlo_lim': {
+        'AgentCount': 'infinity',
+        'T_age': 'infinity'
+    }
+})
+
+# Optional more detailed _fcts about various parameters
+CRRA_fcts = {
+    'about': 'Coefficient of Relative Risk Aversion'}
+CRRA_fcts.update({'latexexpr': r'\providecommand{\CRRA}{\rho}\CRRA'})
+CRRA_fcts.update({'_unicode_': 'ρ'})  # \rho is Greek r: relative risk aversion rrr
+CRRA_fcts.update({'prmtv_par': 'True'})
+init_perfect_foresight_plus['prmtv_par'].append('CRRA')
+# init_perfect_foresight_plus['_fcts'].update({'CRRA': CRRA_fcts})
+init_perfect_foresight_plus.update({'CRRA_fcts': CRRA_fcts})
```

Comment:
Could you remind me what the different fcts are for?
Some seem to be about presentation.

Maybe a more explicit presentation layer is in order.

Reply:
Perhaps a better alternative would be to use a python feature introduced in 3.9:  Variables can now have documentation/explanations directly attached to them.  Basically, seems to allow what I'm doing by adding _fcts, as built-in part of python.  

I've long wondered why programming languages don't all allow this, rather than expecting users to hunt down potential explanations in the code.  No good reason, I think.